File: desktop/entangler/src/main/java/ru/dbudyak/entangler/Side.java
package ru.dbudyak.entangler;

/**
 * Created by dbudyak on 08.06.14.
 */
public class Side {
    private boolean isConnected = false;
    private Direction direction = Direction.NONE;

    public boolean isConnected() {
        return isConnected;
    }

    public void setConnected(boolean isConnected) {
        this.isConnected = isConnected;
    }

    public Direction getDirection() {
        return direction;
    }

    public void setDirection(Direction direction) {
        this.direction = direction;
    }

    public static enum Direction {
        INPUT, OUTPUT, NONE
    }
}

File: desktop/entangler/src/main/kotlin/ru/dbudyak/entangler/Side.kt
package ru.dbudyak.entangler

/**
 * Represents one side of a quantum element (left, top, right, bottom).
 * Each side has a direction (INPUT, OUTPUT, or NONE) and connection status.
 */
class Side {
    var isConnected: Boolean = false
    var direction: Direction = Direction.NONE

    enum class Direction {
        INPUT, OUTPUT, NONE
    }
}
